integration/test_client.py

```diff
@@ -355,6 +357,75 @@ def test_client_with_extra_options() -> None:
         #     auth_credentials=WCS_CREDS,
         #     additional_config=additional_config,
         # ),
-        weaviate.connect_to_embedded(additional_config=additional_config),
+        weaviate.connect_to_embedded(
+            port=8070, grpc_port=50040, additional_config=additional_config
+        ),
     ]:
         assert client._connection.timeout_config == _Timeout(1, 2)
+
+
+def test_connect_and_close_to_embedded() -> None:
+    # Can't use the default port values as they are already in use by the local instances
+    client = weaviate.connect_to_embedded(port=8078, grpc_port=50151, version="1.23.5")
+
+    client.connect()
+    assert client.is_connected()
+    metadata = client.get_meta()
+    assert "1.23.5" == metadata["version"]
+    assert client.is_ready()
+    assert "8078" == metadata["hostname"].split(":")[2]
+    assert client.is_live()
+
+    client.close()
+    assert not client.is_connected()
+    with pytest.raises(WeaviateClosedClientError):
+        client.get_meta()
+
+
+def test_embedded_as_context_manager() -> None:
+    default_version = "1.23.5"
+    with weaviate.connect_to_embedded(port=8077, grpc_port=50152) as client:
+        assert client.is_connected()
+        metadata = client.get_meta()
+        assert default_version == metadata["version"]
+        assert client.is_ready()
+        assert client.is_live()
+
+    assert not client.is_connected()
+    with pytest.raises(WeaviateClosedClientError):
+        client.get_meta()
+
+
+def test_embedded_with_wrong_version() -> None:
+    with pytest.raises(weaviate.exceptions.WeaviateEmbeddedInvalidVersionError):
+        weaviate.connect_to_embedded(version="this_version_does_not_exist")
+
+
+def test_embedded_already_running() -> None:
+    client = weaviate.connect_to_embedded(port=8096, grpc_port=50154)
+    assert client._connection.embedded_db is not None
+    assert client._connection.embedded_db.process is not None
+
+    with pytest.raises(weaviate.exceptions.WeaviateStartUpError):
+        weaviate.connect_to_embedded(port=8096, grpc_port=50154)
+
+    client.close()
+
+
+def test_connect_to_embedded_with_already_running_local_weaviate() -> None:
+    with pytest.raises(weaviate.exceptions.WeaviateStartUpError):
+        weaviate.connect_to_embedded(port=8080, grpc_port=50051)
```

Comment:
how does this work in the CI? There we start tests in a container and afaik there shouldn't be a running weavaite. Also the two test below test the same without relying on something else running. Should we just delete the test?

Reply:
That's a good point, will remove!